backend/src/indexer/block_preprocessor.rs

```diff
@@ -373,15 +437,41 @@ pub async fn compute_validator_staking_information(
         }
     }
 
-    let validators_staking_information: ValidatorStakingInformation = ValidatorStakingInformation {
+    // prepare the validator staking information which will be updated later in the
+    // database
+    let validators_staking_information = ValidatorStakingInformation {
         ids: validator_ids,
         staked_amounts: validator_staked_amounts,
         pool_total_staked_amounts: validator_pool_staked,
     };
 
+    debug!(
+        "computed staking information through client, total staked: {:?}, for height: {:?}, \
+         validator staking ids: {:?}, validator staked amounts: {:?}, validator pool staked: {:?}",
+        &total_staked,
+        &block_height,
+        &validators_staking_information.ids,
+        &validators_staking_information.staked_amounts,
+        &validators_staking_information.pool_total_staked_amounts
+    );
     Ok((total_staked, validators_staking_information))
 }
 
+/// returns the block height from a provided Block Identifier.
+/// We are expecting this to be an AbsoluteBlockHeight where the height can be
+/// taken as u64 (later will be used for caching with computing staked amounts)
+/// we will immediately create an error if its not possible as the indexer
+/// should always work with Absolute height for this area of the code.
+fn get_block_height_from_absolute(block_height: BlockIdentifier) -> u64 {
+    match block_height {
+        v2::BlockIdentifier::AbsoluteHeight(h) => h.height,
+        _ => {
+            debug!("Non-absolute block identifier; skipping cache.");
+            panic!("Expected absolute block height in order to work with caching effectively");
+        }
+    }
+}
```

Comment:
If you create a function then the intention is that the function likely will be re-used at some point in the future and the function at is should be generic. As such the comment should not highlight its specific use at one specific code base point. If you only want the code to be used at that specific context then I would move this function + comment into the place where it is used (rather than creating a separate re-usable function).